c/src/ml-api-service-db.cc

```diff
@@ -0,0 +1,268 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/**
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * @file ml-api-service-db.c
+ * @date 07 Mar 2022
+ * @brief NNStreamer/Service C-API implementation
+ * @see	https://github.com/nnstreamer/nnstreamer
+ * @author MyungJoo Ham <myungjoo.ham@samsung.com>
+ * @author Sangjung Woo <sangjung.woo@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include <iostream>
+#include <cstring>
+
+#include "ml-api-inference-internal.h"
+#include "ml-service-db.hh"
+#include "ml-api-service.h"
+
+#define ML_DATABASE_PATH      SYS_DB_DIR"/.ml-service-leveldb"
+
+/**
+ * @brief Class for implementation of IMLServiceDB
+ */
+class MLServiceLevelDB : public IMLServiceDB
+{
+public:
+  MLServiceLevelDB (std::string path);
+  virtual ~ MLServiceLevelDB ();
+  virtual void connectDB () override;
+  virtual void setPipelineDescription (std::string name,
+      std::string pipeline_description) override;
+  virtual void getPipelineDescription (std::string name,
+      std::string & pipeline_description) override;
+  virtual void delPipelineDescription (std::string name);
+
+private:
+  std::string path;
+  leveldb_t *db_obj;
+  leveldb_readoptions_t *db_roptions;
+  leveldb_writeoptions_t *db_woptions;
+};
+
+static IMLServiceDB *instance = nullptr;
+
+/**
+ * @brief Get the instance object of MLServiceLevelDB
+ * @return IMLServiceDB* instance of MLServiceLevelDB
+ */
+static IMLServiceDB *
+get_instance ()
```

Comment:
Discussion: make instance as static in MLServiceLevelDB class
declare static instance; in class MLServiceLevelDB

MLServiceLevelDB * MLServiceLevelDB::instatnce = nullptr;

MLServiceLevelDB *
MLServiceLevelDB::getInstance ()
{
 ...
}

Reply:
I will apply `Singleton` pattern for this issue.
Thank you for your advice.